kinto/core/resource/__init__.py

```diff
@@ -148,6 +161,14 @@ class UserResource:
     """Schema to validate records."""
 
     def __init__(self, request, context=None):
+
+        self.request = request
+        self.context = context
+
+        # XXX: Check if the path is valid before doing anything. This have to be done
+        # before fetching any model, so we can't wait for regular colander validation.
+        self._raise_400_if_invalid_path()
```

Comment:
Yeah, that's a way to go. My first plan was to set model as a `@property`, but this lead to several recursions to solve and I just gave up for some time hahaha.